### PR TITLE
[WIP] File.open/2: complete list of modes

### PR DESCRIPTION
**Pull request by eksperimental**
I have updated the modes, since many were missing. In docs as well as in typespecs

### CODE REVIEW

File: lib/elixir/lib/file.ex [hunk 1]
```diff
@@ -920,17 +920,19 @@ defmodule File do
...
-  in the `IO` module. By default, a file is opened in binary mode,
+  in the `IO` module. By default, a file is opened in `:binary` mode,
   which requires the functions `IO.binread/2` and `IO.binwrite/2`
   to interact with the file. A developer may pass `:utf8` as an
   option when opening the file and then all other functions from
   `IO` are available, since they work directly with Unicode data.
 
   The allowed modes:
 
+    * `:binary` - the default mode.
```
Comment (eksperimental) [inline lib/elixir/lib/file.ex]: Does anybody have a better description for :binary?

Comment (NobbZ) [inline lib/elixir/lib/file.ex]: Opens the file in binary mode, disabling special handling of unicode sequences (default). 

Something along this line should be appropriate and sufficient.

Comment (josevalim) [inline lib/elixir/lib/file.ex]: :+1:

Comment (eksperimental) [inline lib/elixir/lib/file.ex]: thank you @NobbZ ,.
updated

**Pull request merged**